components/JournalForm.js

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from "react";
+import formatDate from "./TransformDateTime";
+import JournalCard from "./JournalCard";
+import useLocalStorageState from "use-local-storage-state";
+import styled from "styled-components";
+
+export default function JournalForm() {
+  const [entries, setEntries] = useLocalStorageState("journalEntries", []);
+  const [subject, setSubject] = useState("");
+  const [text, setText] = useState("");
+  const [selectedDate, setSelectedDate] = useState(new Date().toISOString());
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const newEntry = {
+      subject,
+      text,
+      date: selectedDate,
+    };
+
+    setEntries((entries) => [...(entries || []), newEntry]);
+    setSubject("");
+    setText("");
+    setSelectedDate(new Date().toISOString());
+  };
+
+  // const { date, time } = formatDate(selectedDate);
+  const handleDelete = (indexToDelete) => {
+    setEntries((entries) =>
+      entries.filter((_, index) => index !== indexToDelete)
+    );
+  };
+
+  return (
+    <>
+      <StyledFormContainer>
+        <FormTitle>Journal Entry</FormTitle>
+        <FormWrapper>
+          <StyledJournalForm onSubmit={handleSubmit}>
+            <section>
+              <Input
+                aria-label="subject"
+                placeholder="Subject"
+                id="subject"
+                type="text"
+                value={subject}
+                onChange={(event) => setSubject(event.target.value)}
+              />
+            </section>
+
+            <StyledDateAndTimeInput
+              aria-label="date-time"
+              id="date-time"
+              name="dateTime"
+              type="datetime-local"
+              value={selectedDate.slice(0, 16)}
+              onChange={(event) => setSelectedDate(event.target.value)}
+            />
+
+            <section>
+              <Textarea
+                aria-label="textarea"
+                placeholder="How are you today?"
+                id="text"
+                value={text}
+                onChange={(event) => setText(event.target.value)}
+              />
+            </section>
+
+            <SubmitButton aria-label="submit-button" type="submit">
+              Add Entry
+            </SubmitButton>
+          </StyledJournalForm>
+        </FormWrapper>
+      </StyledFormContainer>
+
+      <CardWrapper>
+        {entries && entries.length > 0 ? (
+          entries.map((entry, index) => (
+            <JournalCard
+              key={index}
+              journalEntry={entry}
+              onDelete={() => handleDelete(index)}
+            />
+          ))
+        ) : (
+          <StyeldText>No entries yet</StyeldText>
+        )}
+      </CardWrapper>
```

Comment:
I chose to keep the JournalCard rendering in JournalForm because this component manages the entries state using useLocalStorageState. Since both creating and displaying entries are closely related in this context, separating them into another component felt unnecessary for the current scope of the project.

Reply:
You can leave it if you want but to give a good review I have to mention this:
It is NOT a good practice. What you say: You have the entries state here inside the form. But this IS the problem. If you ever will render a journal entry in your app you will have to render the form (this feels unnecessary). A second reason is: A component should always be as thumb as possible (pure function). The component should only operate on the props it receives. This paradigm will prevent a lot of bugs. All the calculating stuff should happen on `page-level` and pass the result of the calculation as props to the component. 
(Keep this in mind for further coding and improvements)